storagenode/piecestore/endpoint.go

```diff
@@ -61,10 +63,11 @@ type Endpoint struct {
 }
 
 // NewEndpoint creates a new piecestore endpoint.
-func NewEndpoint(log *zap.Logger, signer signing.Signer, trust *trust.Pool, store *pieces.Store, pieceinfo pieces.DB, orders orders.DB, usage bandwidth.DB, usedSerials UsedSerials, config Config) (*Endpoint, error) {
+func NewEndpoint(log *zap.Logger, signer signing.Signer, trust *trust.Pool, store *pieces.Store, pieceinfo pieces.DB, orders orders.DB, usage bandwidth.DB, usedSerials UsedSerials, config Config, oldConfig psserver.Config) (*Endpoint, error) {
```

Comment:
Maybe pass in `monitor.Service` into the endpoint and use the `monitor.Service` to handle the space and bandwidth logic?

Reply:
yeah, now when I'm checking space and bandwidth for upload it makes more sense